js/modules/k6/execution/execution.go

```diff
@@ -314,13 +317,8 @@ func (o *tagsDynamicObject) Get(key string) goja.Value {
 // representation. An exception is raised in case a denied type is provided.
 func (o *tagsDynamicObject) Set(key string, val goja.Value) bool {
 	var err error
-	o.state.Tags.Modify(func(tags *metrics.TagSet) *metrics.TagSet {
-		newTags, applyErr := common.ApplyCustomUserTag(o.runtime, tags, key, val)
-		if applyErr != nil {
-			err = applyErr
-			return tags
-		}
-		return newTags
+	o.state.Tags.Modify(func(tagsAndMeta *metrics.TagsAndMeta) {
+		err = common.ApplyCustomUserTag(o.runtime, tagsAndMeta, key, val)
```

Comment:
just a nitpick, if we panic here, we can keep the scopes cleaner:

like:
```
o.state.Tags.Modify(func(tagsAndMeta *metrics.TagsAndMeta) {
		if err := common.ApplyCustomUserTag(o.runtime, tagsAndMeta, key, val); err != nil {
            panic(o.runtime.NewTypeError(err.Error()))
        }
}

return true
```

Reply:
I am not sure if the lambda won't be shown in the JS stracktrace if we panic directly from it? :thinking:

Reply:
Added in this fixup because it seems fine to me https://github.com/grafana/k6/pull/2726/commits/2428c114bbe91aff3e1685e6f47f1453aa561949